### PR TITLE
feat: show recogized text content

### PR DESCRIPTION
**Pull request by ronnymikalsen**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalLibraryOfNorway/ngx-mime/blob/main/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Recognized text do not to appear when alto is available

Issue Number: N/A

## What is the new behavior?
To make it easier to read content when handwriting or font is difficult to read
As a person
Then I want the recognized text to appear

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  });
+
+  it('should load current alto on canvas change', fakeAsync(() => {
+    service.initialize();
+    iiifManifestService.load('fakeUrl').subscribe(() => {
+      waitForDebounce();
+      mockAltoRequest();
+      expectAltoToBeDefined();
+    });
+  }));
```
Comment (raymondkarstensen) [inline libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts:56]: From test description I would expect a call to a canvas change function in the test, but cannot find one.

**Review by raymondkarstensen — CHANGES_REQUESTED** [review]

File: libs/ngx-mime/src/lib/core/builders/alto/strings.builder.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    return this;
+  }
+
+  build(): String {
+    return this.stringXml
+      ? this.stringXml.map((string: any) => {
+          return { content: string.$.CONTENT };
+        })
+      : [];
+  }
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/builders/alto/strings.builder.ts]: That is correct. It also is missing correct import of String. I have fixed it

File: libs/ngx-mime/src/lib/core/builders/alto/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { AltoBuilder } from './alto.builder';
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/builders/alto/index.ts:1]: None of the others builders are used outside of the current directory.

File: libs/ngx-mime/src/lib/core/intl/viewer-intl.lt.ts [hunk 4]
```diff
@@ -36,6 +36,8 @@ export class MimeViewerIntlLt extends MimeViewerIntl {
   manifestUriMissingLabel = 'Nerastas objektų sąrašo identifikatorius (ManifestUri)';
   manifestNotValidLabel = 'Netinkamas objektų sąrašas (Manifest)';
   pageDoesNotExists = 'Nepavyko rasti šio paslapio';
+  textErrorLabel = 'Atsiprašau, nerandu jums teksto';
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/intl/viewer-intl.lt.ts:41]: I think both translations may be correct. 
https://translate.google.com/?sl=auto&tl=no&text=Atsipra%C5%A1au%2C%20nerandu%20jums%20teksto&op=translate

[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/intl/viewer-intl.lt.ts:41]: 'Atsiprašau, bet nerandu jums teksto' is mayby a little more polite?

File: libs/ngx-mime/src/lib/core/intl/viewer-intl.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ export class MimeViewerIntl {
   closeLabel = 'Close';
   attributionLabel = 'Attribution';
   attributonCloseAriaLabel = 'Close attribution dialog';
+  textLabel = 'Show text';
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/intl/viewer-intl.ts]: Fixed and refactored to recognizedTextLabel

File: libs/ngx-mime/src/lib/core/mime-viewer-config.ts [hunk 6]
```diff
@@ -88,6 +90,10 @@ export class MimeViewerConfig {
         fields.isDropEnabled !== undefined
           ? fields.isDropEnabled
           : this.isDropEnabled;
+      this.initTextContentToggle =
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/mime-viewer-config.ts]: fixed

File: libs/ngx-mime/src/lib/core/intl/viewer-intl.lt.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/intl/viewer-intl.lt.ts]: Changed to 'Atsiprašau, bet nerandu jums teksto'

File: libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts [hunk 7]
```diff
@@ -67,11 +68,13 @@ export class ViewerHeaderComponent implements OnInit, OnDestroy {
   mimeHeaderAfter!: ViewContainerRef;
   public manifest: Manifest | null = null;
   public state = 'hide';
+  textContentToggle = false;
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.ts]: Attribute not used anymore, so I have deleted it

File: libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.html [hunk 8]
```diff
@@ -14,6 +14,16 @@
       fxLayoutAlign="end center"
       class="buttons-container"
     >
+      <mat-button-toggle
+      *ngIf="hasAlto"
+      id="mime-toggleTextButton"
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/viewer/viewer-header/viewer-header.component.html]: fixed

File: apps/integration-e2e/features/recognized-text-content.feature [hunk 9]
```diff
@@ -0,0 +1,15 @@
+Feature: Show recogized text content
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/integration-e2e/features/recognized-text-content.feature]: Fixed

File: apps/integration-e2e/features/recognized-text-content.feature [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  Scenario: Enable text content display
+    Given the viewer is opened with a publication with recognized text
+    When the viewer is in dashboard view
+    Then the user should be able to enable text content display
+
+  Scenario: Show text content
+    Given the viewer is opened with a publication with recognized text
+    And the viewer is in dashboard view
+    When the user enables text content display
+    Then the text content should be shown
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/integration-e2e/features/recognized-text-content.feature]: Fixed

File: apps/integration-e2e/pages/viewer.po.ts [hunk 11]
```diff
@@ -145,6 +156,29 @@ export class ViewerPage {
     }
   }
 
+  async haveRecognizedTextToggle(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/integration-e2e/pages/viewer.po.ts:159]: Maybe  isRecognizedTextContentToggleButtonPresent is even better :-). Fixed

File: apps/integration-e2e/pages/viewer.po.ts [hunk 12]
```diff
@@ -127,11 +135,14 @@ export class ViewerPage {
       by.css('#toggleSinglePageViewButton')
     );
     this.twoPageViewButtonEl = element(by.css('#toggleTwoPageViewButton'));
+    this.recognizedTextDisplayToggleEl = element(by.css('#mime-toggleRecognizedTextButton'));
```
[the remark this answers is not in the document]
Comment (ronnymikalsen) [reply apps/integration-e2e/pages/viewer.po.ts]: Renamed to 'recognizedTextContentToggleButtonEl'

File: libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts (hunk 1 above)
Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts]: Fixed. Changing to canvas index 2.

Comment (ronnymikalsen) [reply libs/ngx-mime/src/lib/core/alto-service/alto.service.spec.ts]: I have changed the test description and added another test for canvas change

**Pull request merged**